torch/futures/__init__.py

```diff
@@ -85,7 +104,7 @@ def set_result(self, result):
         super().set_result(result)
 
 
-def collect_all(futures):
+def collect_all(futures: List[Future[T]]) -> Future[List[Future[T]]]:
```

Comment:
IIUC, we don't expect all Futures here contains the type T object. Do we need to use Any here? cc @jjlilley

Reply:
Updated typing decorator not to use generics here.